plugins/jenkins-backend/src/service/jenkinsApi.ts

```diff
@@ -98,9 +105,25 @@ export class JenkinsApiImpl {
         // at all.
         // Whitespaces are only included for readability here and stripped out
         // before sending to Jenkins
-        tree: JenkinsApiImpl.jobsTreeSpec.replace(/\s/g, ''),
+        tree: treeSpec,
       });
 
+      if (jenkinsInfo.isLatestCICDBuildsEnabled && project.jobs) {
+        project.jobs = project.jobs
+          .sort((jobA: any, jobB: any) => {
+            // sort builds with no last build to the end
+            if (jobA?.lastBuild === null) {
+              return 1;
+            }
+            if (jobB?.lastBuild === null) {
+              return -1;
+            }
+
+            return jobB.lastBuild.timestamp - jobA.lastBuild.timestamp;
+          })
+          .slice(0, 50);
```

Comment:
Thanks for the review. Sorry, I should have provide more context. 

>Why is this sorting happening here only for the isLatestCICDBuildsEnabled? Is the frontend sorting added in the mentioned PR (Sort Jenkins biuld by timestemp in CITable #7594) not enough?

The jenkins client sends an api request here: [https://github.com/backstage/backstage/blob/master/plugins/jenkins-backend/src/service/jenkinsApi.ts#L108](https://github.com/backstage/backstage/blob/master/plugins/jenkins-backend/src/service/jenkinsApi.ts#L108). The tree basically describes how the response will be formatted [1] . At the end of the url, it is specified to always return 50 builds: [https://github.com/backstage/backstage/blob/master/plugins/jenkins-backend/src/service/jenkinsApi.ts#L65](https://github.com/backstage/backstage/blob/master/plugins/jenkins-backend/src/service/jenkinsApi.ts#L65). Now, this might be fine depending on the configurations of jenkins [3]. For us, some projects contain more than 50 builds because of configurations we have. Based on my tests, the 50 builds returned by jenkins are not actually very useful. I spend time reading up on XPath/jenkins api, but I could not figured out how to construct an url to retrieve the last 50 builds (might not be possible), hence I decided to retrieve all builds and just sort them before creating the response. This is slower, hence the opt-in configuration.

>Why do you slice the jobs to 50 eventhough you have loaded all of them?

Basically, decided to keep the last 50 jobs similar to the current behaviour instead of responding with all builds. I do not see a reason to return all builds, and the ui seems a bit slower with lots of builds.

Hope this clarifies things :pray:

[1] [https://www.jenkins.io/doc/book/using/remote-access-api/#RemoteaccessAPI-XPathselection](https://www.jenkins.io/doc/book/using/remote-access-api/#RemoteaccessAPI-XPathselection)

[2] [https://stackoverflow.com/questions/47930226/how-to-limit-jenkins-api-response-to-last-n-build-ids](https://stackoverflow.com/questions/47930226/how-to-limit-jenkins-api-response-to-last-n-build-ids)

[3] [https://stackoverflow.com/questions/69100613/how-to-discard-old-builds-in-jenkins-based-on-their-completion-status](https://stackoverflow.com/questions/69100613/how-to-discard-old-builds-in-jenkins-based-on-their-completion-status)

Reply:
Thanks for the context & explanation - appreciate it! Makes me think if the limit to 50 should be a config variable/optional as well, but agree that it's out of the scope of the PR & that it's fine to keep the current behaviour. Looks good to me - let's get another reviewer in & than it should be good to go.

Reply:
Thanks for the review @tudi2d ! Just curious, related to `let's get another reviewer in & than it should be good to go`, is there a process for this. Should I ping somebody in discord or somebody will have a look whenever they have time.